src/logger/httpLoggerRules_test.go

```diff
@@ -286,195 +286,662 @@ func TestUsesRemoveRules(t *testing.T) {
+	httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
+	assert.Equal(t, 1, len(queue), "queue length is not 1")
+	assert.Equal(t, true, strings.Contains(queue[0], "[\"response_body\",\"<html>Hello xyxy!</html>\"],"), "custom string not replaced in queue")
+
+	request = helper.MockRequestWithJson2()
+	mockResponse = helper.MockResponseWithHtml()
+
+	request.Body = ioutil.NopCloser(bytes.NewBufferString(helper.mockJson))
+	mockResponse.Body = ioutil.NopCloser(bytes.NewBufferString(helper.mockHtml))
+	mockResponse.Request = request
+
+	queue = make([]string, 0)
+	logger = NewHttpLoggerQueueRules(queue, "!response_body! replace !World!, !<b>$&</b>!")
+	httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
+	assert.Equal(t, 1, len(queue), "queue length is not 1")
+	assert.Equal(t, true, strings.Contains(queue[0], "[\"response_body\",\"<html>Hello <b>World</b>!</html>\"],"), "custom string not replaced in queue")
+
+	queue = make([]string, 0)
+	logger = NewHttpLoggerQueueRules(queue, "!response_body! replace !(World)!, !<b>$1</b>!")
+	httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
+	assert.Equal(t, 1, len(queue), "queue length is not 1")
+	assert.Equal(t, true, strings.Contains(queue[0], "[\"response_body\",\"<html>Hello <b>World</b>!</html>\"],"), "custom string not replaced in queue")	
+
+	request = helper.MockRequestWithJson2()
+	mockResponse = helper.MockResponseWithHtml()
+
+	request.Body = ioutil.NopCloser(bytes.NewBufferString(helper.mockJson))
+	mockResponse.Body = ioutil.NopCloser(bytes.NewBufferString(helper.mockHtml5))
+	mockResponse.Request = request
+
+	queue = make([]string, 0)
+	logger = NewHttpLoggerQueueRules(queue, "!response_body! replace !<input([^>]*)>([^<]*)</input>!, !<input$1></input>!")
+	httpMessage.sendNetHttpRequestResponseMessage(logger, mockResponse, 0, 0)
+	assert.Equal(t, 1, len(queue), "queue length is not 1")
+	assert.Equal(t, true, strings.Contains(queue[0], "[\"response_body\",\"<html>\\n<input type=\\\"hidden\\\"></input>\\n<input class='foo' type=\\\"hidden\\\"></input>\\n</html>\"],"), "custom string not replaced in queue")
+}
+
 // test uses sample rules
 
+func TestUsesSampleRules(t *testing.T) {
+	helper := GetTestHelper()
+
+	request := helper.MockRequestWithJson2()
+	mockResponse := helper.MockResponseWithHtml()
+
+	request.Body = ioutil.NopCloser(bytes.NewBufferString(helper.mockJson))
+	mockResponse.Body = ioutil.NopCloser(bytes.NewBufferString(helper.mockHtml))
+	mockResponse.Request = request
+	
+	queue := make([]string, 0)
+
+	logger, err := NewHttpLoggerQueueRules(queue, "sample 10\nsample 99")
+	if err != nil {
+		assert.Equal(t, "Multiple sample rules", err.message, "multiple sample rule error not correct") //This is what I came up with as Go has no Try & Catch functionality for errors
+	}
+	
+	logger = NewHttpLoggerQueueRules(queue, "sample 10")
```

Comment:
if `NewHttpLoggerQueueRules` return `(*HttpLogger, error)` it will always return it so you need to catch the error: `logger, _ := NewHttpLoggerQueueRules(queue, "sample 10")` you may or may not need the `:`